itest/rust/src/engine_tests/match_class_test.rs

```diff
@@ -164,3 +164,24 @@ fn match_class_control_flow() {
 
     assert!(broken, "break statement should have been executed");
 }
+
+#[itest]
+fn match_class_unit_type() {
+    let obj: Gd<Object> = Object::new_alloc();
+    let to_free = obj.clone();
+    let mut val = 0;
+
+    let _: () = match_class! { obj,
```

Comment:
```suggestion
    match_class! { obj,
```
To test a more natural usage, and make sure this works without type inference.